### DISCUSSION TITLE
sslstrip.py doesn't work

### DISCUSSION THREAD

**Discussion by ifoolb** [mitmproxy/mitmproxy#5680] [Q&A] [answered]
Hello there, I run mitmdump -s sslstrip and set http proxy to my server, port 8080 in Firefox, then I visit http://cat.com:80, but it's still redirecting to https:
127.0.0.1:56472: GET http://cat.com/
              << 301 Moved Permanently 0b
127.0.0.1:56472: clientdisconnect
127.0.0.1:56474: clientconnect
127.0.0.1:56474: CONNECT www.cat.com:443
 << Cannot establish TLS with client (sni: www.cat.com): TlsException("SSL handshake error: Error([('SSL routines', 'ssl3_read_bytes', 'sslv3 alert certificate unknown')])")

while http only sites such as httpforever.com work correctly. Is there a way to let mitmproxy strip out ssl and just use plain HTTP to communicate with the client?

**Comment by Prinzhorn** [comment, marked as answer]
From my comment below:

> Browsers have HSTS for bing.com hardcoded, it's impossible to strip ssl (at least to my knowledge) https://hstspreload.org/?domain=bing.com

---

Have you visited the site before or is it a clean Firefox with no history etc. (clean profile, private window)?

Works for me with 8.1.1 via `./mitmproxy --scripts sslstrip.py`. What version are you using? You shouldn't get an SSL error, do you get that without the addon as well? All other logs look as expected, mitmproxy still needs to connect to the website over SSL.

![Screenshot from 2022-10-27 12-22-48](https://user-images.githubusercontent.com/679144/198260210-dc30f9dc-1664-4965-b8e3-891097a16adf.png)

**Reply by ChillVibesMushroom** [reply]
> From my comment below:
> 
> > Browsers have HSTS for bing.com hardcoded, it's impossible to strip ssl (at least to my knowledge) https://hstspreload.org/?domain=bing.com
> 
> Have you visited the site before or is it a clean Firefox with no history etc. (clean profile, private window)?
> 
> Works for me with 8.1.1 via `./mitmproxy --scripts sslstrip.py`. What version are you using? You shouldn't get an SSL error, do you get that without the addon as well? All other logs look as expected, mitmproxy still needs to connect to the website over SSL.
> 
> ![Screenshot from 2022-10-27 12-22-48](https://user-images.githubusercontent.com/679144/198260210-dc30f9dc-1664-4965-b8e3-891097a16adf.png)
 
 

you’re saying we have to use a certain version and wipe the history from the browser for this to work in stripping ssl when I do this some sites like eBay won’t strip is there another way to use sslstrip ? I have the bettercap certificates

**Reply by Prinzhorn** [reply]
HSTS was invented to mitigate ssl strip attacks. So if you've already visited a website before (or HSTS is hard coded into the browser) it does exactly what it's supposed to and prevents the attack.

**Comment by ifoolb** [comment]
Ok, cat.com (a random domain I typed) does work, after clicking through a browser warning page. But bing.com still doesn't work, and as before, the warning page doesn't allow me to proceed:

"You can't visit www.bing.com right now because the website uses HSTS. Network errors and attacks are usually temporary, so this page will probably work later."

And I was testing in private mode, but even freshly installed Chromium doesn't work with sites that have HSTS enabled such as Bing

**Reply by Prinzhorn** [reply]
Browsers have HSTS for bing.com hardcoded, it's impossible to strip ssl (at least to my knowledge) https://hstspreload.org/?domain=bing.com
